services/QuillLMS/app/workers/teacher_activity_feed_refill_worker.rb

```diff
@@ -0,0 +1,36 @@
+# This worker clears the content of the TeacherActivityFeed
+# and refills it will the teacher's most recent 40 activity sessions
+class TeacherActivityFeedRefillWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: 'critical'
+
+  def perform(user_id)
+    teacher = Teacher.find(user_id)
+
+    return unless teacher
+
+    ids = ids_for_activity_feed(teacher)
+
+    return if ids.empty?
+
+    TeacherActivityFeed.reset!(user_id)
+    # Note, reversing id order since .add
+    # pops each from the front of the array
+    TeacherActivityFeed.add(user_id, ids.reverse)
```

Comment:
I feel like a feed is more LIFO than FIFO.  Would `rpush` here:  https://github.com/empirical-org/Empirical-Core/blob/feat/redis_feed/services/QuillLMS/app/models/redis_feed.rb#L42 make more sense and also allow future self to not have to remember to reverse items?

Reply:
So this is suppose to be a LIFO queue, just that redis handles adding a single item to the front of the queue differently than an array. If we were to use `rpush`, you have to work around trimming to a size (there’s no `rtrim` only `ltrim`). Given that we do typically want newer items first in the list of a feed, I like the idea of adding to the beginning rather than the end, since that prevents having to do a `reverse` somewhere else. But honestly, either way would work. \

Given this, feedback, I think I might push the `reverse` complexity into my RedisFeed `.add` code so the usage will feel more obvious, so that adding an item or an array will have the same functionality.

Reply:
I've updated the interface so that an array slides onto the front of the queue maintaining its order. I agree that would be more expected than having to remember to reverse the items.

This also led to simplifying the `callback_on_add(id)` to only have to handle a single id at a time (it'll be called for each item of the array). I think this will be easier to reason around.